src/Game/zSaveLoad.cpp

```diff
@@ -2,41 +2,180 @@
+extern uint32 gFrameCount;
+extern float32 sTimeElapsed;
+extern long32 sTimeLast;
+extern long32 sTimeCurrent;
+
 // func_800AD20C
 #pragma GLOBAL_ASM("asm/Game/zSaveLoad.s", "zUpdateThumbIcon__Fv")
 
 // func_800AD328
 #pragma GLOBAL_ASM("asm/Game/zSaveLoad.s", "zSaveLoad_Tick__Fv")
 
+
 // func_800AD598
 #pragma GLOBAL_ASM("asm/Game/zSaveLoad.s", "zSaveLoad_poll__Fi")
 
+#ifndef NON_MATCHING
 // func_800AD600
 #pragma GLOBAL_ASM("asm/Game/zSaveLoad.s", "zSendEventToThumbIcon__FUi")
-
+#else
+void zSendEventToThumbIcon(uint32 p) {
+    char* iconString = "MNU3 THUMBICON";
+    if (gGameMode == eGameMode_Load)
+    {
+        iconString = "MNU4 THUMBICON";
+    }
+    zEntEvent(zSceneFindObject(xStrHash(iconString)), p);
+}
+#endif
+
+#ifndef NON_MATCHING
 // func_800AD654
 #pragma GLOBAL_ASM("asm/Game/zSaveLoad.s", "zChangeThumbIcon__FPCc")
+#else
+void zChangeThumbIcon(const char* icon)
+{
+    char* iconString;
+    float32 arr[4];
+
+    memset(arr, 0, sizeof(arr));
+    arr[0] = xStrHash(icon);
+    
+    iconString = "MNU3 THUMBICON";
+    if (gGameMode == eGameMode_Load)
+    {
+        iconString = "MNU4 THUMBICON";
+    }
+    zEntEvent(zSceneFindObject(xStrHash(iconString)), 0x1f5, arr);
+}
+#endif
 
 // func_800AD6C8
-#pragma GLOBAL_ASM("asm/Game/zSaveLoad.s", "zSaveLoadInit__Fv")
+void zSaveLoadInit()
+{
+    zSaveLoadUITableInit(zSaveLoadUITable);
+    zSaveLoadGameTableInit(zSaveLoadGameTable);
+}
 
 // func_800AD6FC
-#pragma GLOBAL_ASM("asm/Game/zSaveLoad.s", "zSaveLoadGameTableInit__FP13zSaveLoadGame")
+void zSaveLoadGameTableInit(zSaveLoadGame* saveTable)
+{
+    for(int32 i = 0; i < 3; i++)
+    {
+        saveTable[i].label[0] = 0;
+        saveTable[i].date[0] = 0;
+        saveTable[i].progress = 0;
+        saveTable[i].size = 0;
+        saveTable[i].thumbIconIndex = 0;
+    }
+}
 
 // func_800AD740
-#pragma GLOBAL_ASM("asm/Game/zSaveLoad.s", "zSaveLoadUITableInit__FP11zSaveLoadUI")
+void zSaveLoadUITableInit(zSaveLoadUI* saveTable)
+{
+    //Doesn't match the zSaveLoadUITable size for some reason
+    for(int32 i = 0; i < 61; i++)
+    {
+        saveTable[i].nameID = xStrHash(saveTable[i].name);
+    }
+}
 
 // func_800AD790
 #pragma GLOBAL_ASM("asm/Game/zSaveLoad.s", "zSaveLoad_UIEvent__FiUi")
 
+#if 1
 // func_800AD7D8
 #pragma GLOBAL_ASM("asm/Game/zSaveLoad.s", "zSaveLoadSGInit__F16en_SAVEGAME_MODE")
+#else
+void zSaveLoadSGInit(en_SAVEGAME_MODE mode)
+{
+    if (mode == XSG_MODE_LOAD) {
+        zSaveLoad_UIEvent(0x27, 0x51);
+    }
+    else
+    {
+        zSaveLoad_UIEvent(0x28,0x51);
+    }
+
+    for(int32 i = 0; i < 0x321; i++)
+    {
+        int32 j = i / 200 + (j >> 0x1f);
+        if(j == (i - (i >> 0x1f)) * 200)
+        {
+            zSaveLoad_Tick();
+        }
+    }
+}
+#endif
 
 // func_800AD874
-#pragma GLOBAL_ASM("asm/Game/zSaveLoad.s", "zSaveLoadSGDone__FP17st_XSAVEGAME_DATA")
+void zSaveLoadSGDone(st_XSAVEGAME_DATA* data)
+{
+    if (data->mode == XSG_MODE_LOAD) {
```

Comment:
Make sure that every C++/Header file that you modify is formatted according to the clang-format rules in the root of this repo. VSCode has a clang-format extension so you can have it run every time you save a file, or you can format them before you commit, but we're trying to keep all of the code formatted consistently